models/classes/class.QtiTestService.php

```diff
@@ -873,19 +875,46 @@ private function deleteTestsFromClassByLabel(
         core_kernel_classes_Class $itemClass
     ): void {
         $testService = $this->getTestService();
-        $itemTreeService = $this->getItemTreeService();
+
+        $deletedTests = [];
+        $deletedItemClasses = [];
+        $refs = [];
 
         foreach ($testClass->getInstances() as $testInstance) {
             if ($testInstance->getLabel() === $testLabel) {
+                $deletedTests[] = $testInstance->getUri();
                 $testService->deleteTest($testInstance);
             }
         }
 
         foreach ($itemClass->getSubClasses() as $subClass) {
             if ($subClass->getLabel() === $itemsClassLabel) {
-                $itemTreeService->deleteClass($subClass);
+                $deletedItemClasses[] = $subClass->getUri();
+                $refs = array_merge($refs, $this->deleteItemClass($subClass));
             }
         }
+
+        $this->getEventManager()->trigger(
+            new QtiTestDeletedEvent($deletedTests, $deletedItemClasses, $refs)
+        );
+    }
+
+    private function deleteItemClass(core_kernel_classes_Class $subClass): array
+    {
+        $resourceReferences = [];
+
+        foreach ($subClass->getInstances(true) as $rdfItem) {
+            $qtiItem = $this->getQtiItemService()->getDataItemByRdfItem($rdfItem);
```

Comment:
if yorue goig to call the getQtiItemService() multiple times wouldnt it be better to call it just once outside the loop? :)

Reply:
Already changed to get the ItemTree and Test services just once.